types/react-intl/index.d.ts

```diff
@@ -28,10 +28,8 @@ declare namespace ReactIntl {
         withRef?: boolean;
     }
 
-    type ComponentConstructor<P> = React.ComponentClass<P> | React.StatelessComponent<P>;
-
-    function injectIntl<P>(component: ComponentConstructor<P & InjectedIntlProps>, options?: InjectIntlConfig):
-        React.ComponentClass<P> & { WrappedComponent: ComponentConstructor<P & InjectedIntlProps> };
+    function injectIntl<P extends InjectedIntlProps>(component: React.ComponentType<P>, options?: InjectIntlConfig):
```

Comment:
Seems like it'd be better to make this extend `Partial<InjectedIntlProps>` so that users never have to specify this on their own, but in case they ever do, they have to match those in `InjectedIntlProps`.

Reply:
That seems to go against the whole idea of `injectIntl`. If I want to use `intl`, I should specify it in my props... or not use the HOC.

Reply:
If we look at the modified test, the SFC doesn't type its props but if it did, it would have to specify that it accepts `intl` or it would generate a TS error.

Meaning, the following should be equivalent:

```ts
injectIntl<SomeComponentProps & InjectedIntlProps>(({
    intl: {
        formatDate,
        formatHTMLMessage,
         formatNumber,
         formatMessage,
         formatPlural,
         formatRelative,
         formatTime,
         locale,
         defaultLocale
     },
     className
 }: SomeComponentProps & InjectedIntlProps) => ...
);
```

Reply:
> If I want to use `intl`, I should specify it in my props... or not use the HOC.

But the idea is that by extending `InjectedIntlProps`, you're forcing users to say they use *all* of the props, even if they're only using a few. If you force the type to extend `Partial<InjectedIntlProps>`, they only need to specify some.

Anyway, it doesn't really matter and you're right - I just realized that `InjectedIntlProps` only has a single property anyway so this should be fine.